src/bika/lims/browser/fields/aranalysesfield.py

```diff
@@ -274,11 +275,13 @@ def add_analysis(self, instance, service, **kwargs):
 
         if not analyses:
             # Create the analysis
-            new_id = self.generate_analysis_id(instance, service)
```

Comment:
Because it was non-sense to always do this dance before creating or copying an analysis. So I simply [moved this logic into the `create_analysis` function](https://github.com/senaite/senaite.core/pull/2120/files#diff-7e06c4886a87d3378dd9a03774e3a66f5d8350e6b9de28c8b815b439d0bacf79R77-R81) and removed the generation of the id before analysis creation from here and from [other places as well](https://github.com/senaite/senaite.core/pull/2120/files#diff-7e06c4886a87d3378dd9a03774e3a66f5d8350e6b9de28c8b815b439d0bacf79L404-L409)